### PR TITLE
feat: handle yat addresses for USDC

### LINKED ISSUE #4150: Bug: unable to send USDC to y.at with mapped USDC address
### Overview

Yat enables users to map both an Ethereum address and/or a USDC address to a yat.

Currently, if a yat has no Ethereum address mapped, but has a USDC address mapped, the app will not detect their USDC address when attempting to send USDC to the yat.

Similarly, if a yat has an Ethereum address mapped and a _different_ address mapped to USDC, the app would incorrectly resolve their ETH address.

### References and additional details

Steps to repro:
1. [Confirm that 👶🎵⛰👶 has 0xeA9f460E8d2Aa4a5a4b291ef093df1e17dfDcAAC mapped for USDC](https://a.y.at/emoji_id/%F0%9F%91%B6%F0%9F%8E%B5%E2%9B%B0%F0%9F%91%B6)
2.  Attempt sending USDC to 👶🎵⛰👶 and see that it fails

1. [Confirm that 🦊🚀🌈 has 0x05A1ff0a32bc24265BCB39499d0c5D9A6cb2011c mapped for ETH and 0xD4f520a44cdB0f123108b187Fac9D0009104f8e9 mapped for USDC](https://a.y.at/emoji_id/%F0%9F%A6%8A%F0%9F%9A%80%F0%9F%8C%88)
2. Attempt sending USDC to and see that it resolves to the ETH address

### Acceptance Criteria

AC:
1. If a yat has a USDC address mapped, attempts to send USDC resolve to the mapped USDC address
2. If a yat has no USDC address mapped, attempts to send USDC resolve to the ETH address if one exists

### Need By Date

_No response_

### Screenshots/Mockups

telegram-cloud-photo-size-1-4979010833426262823-x

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

y.at can map different addresses for different assets

this PR handles sends for USDC to a different address than other assets

see #4150 for a good description

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #4150

## Risk

medium - affects all sending

## Testing

**see #4150 for testing steps with y.at addresses**

do a 0 value tx for the following

* ensure unstoppable domains still resolve
* ensure plain eth addresses still resolve
* ensure y.at addresses still resolve
  * ensure sending USDC to the above yats resolve correctly (0 amount)
  * ensure sending ETH to the above yats resolve correctly (0 amount)
* ensure btc addresses still resolve

### Engineering

as above

### Operations

as above

## Screenshots (if applicable)

n/a

### CODE REVIEW

File: src/lib/address/yat.ts [hunk 1]
```diff
@@ -37,19 +39,43 @@ export const validateYat: ValidateYat = ({ value }) => {
...
+          return maybeETHData.data.split('|')[0]
+        }
+      }
+    })()
 
-    if (!found) return ''
-    // data format: address|description|signature|default
-    return toChecksumAddress(found.data.split('|')[0])
+    if (!maybeAddress) return ''
+    return toChecksumAddress(maybeAddress)
```
Comment (gomesalexandre) [inline src/lib/address/yat.ts:78]: Not introduced by this PR but `web3-utils` is only imported for the purpose of getting checksummed addresses, let's remove the dep and use `ethers.utils.getAddress()` ?

File: src/lib/address/yat.ts [hunk 2]
```diff
@@ -37,19 +39,43 @@ export const validateYat: ValidateYat = ({ value }) => {
...
+    const ETH_TAG_ID = '0x1004'
+    const maybeUSDCData = data.result.find(emoji => emoji.tag === USDC_TAG_ID)
+    const maybeETHData = data.result.find(emoji => emoji.tag === ETH_TAG_ID)
+
+    const USDC_ASSET_ID: AssetId = 'eip155:1/erc20:0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
+
+    const maybeAddress = (() => {
+      switch (assetId) {
+        case USDC_ASSET_ID: {
+          if (!maybeUSDCData) return ''
```
Comment (gomesalexandre) [inline src/lib/address/yat.ts:65]: We may want to fallback to the ETH address:
> If a yat has no USDC address mapped, attempts to send USDC resolve to the ETH address if one exists

**Review by gomesalexandre — APPROVED** [review]
Tested locally, confirmed this does what it says on the box:

- Sending USDC sends to the mapped USDC address
- Sending ETH sends to the mapped ETH address

Added one comment WRT the ACC but conceptually looks good in its current state

File: src/lib/address/yat.ts (hunk 2 above)
Comment (0xdef1cafe) [reply src/lib/address/yat.ts:65]: ahhh good point

Comment (0xdef1cafe) [reply src/lib/address/yat.ts]: https://github.com/shapeshift/web/pull/4153/commits/4f825ed63d0eb90baf3b61f5034ef9a4069f1ea3

File: src/lib/address/yat.ts (hunk 1 above)
Comment (0xdef1cafe) [reply src/lib/address/yat.ts]: https://github.com/shapeshift/web/pull/4153/commits/7728d646f4ab599cfcd64b96481b2467e532118f

**Review by gomesalexandre — APPROVED** [review]
in.gif

**Pull request merged**